presto-main/src/test/java/com/facebook/presto/cost/TestJoinStatsRule.java

```diff
@@ -246,6 +247,20 @@ public void testStatsForLeftAndRightJoin()
         assertJoinStats(RIGHT, RIGHT_JOIN_COLUMN, RIGHT_OTHER_COLUMN, LEFT_JOIN_COLUMN, LEFT_OTHER_COLUMN, RIGHT_STATS, LEFT_STATS, leftJoinStats);
     }
 
+    @Test
+    public void testLeftJoinMissingStats()
+    {
+        PlanNodeStatsEstimate leftStats = planNodeStats(
+                0,
+                new SymbolStatistics(LEFT_JOIN_COLUMN, UNKNOWN_STATS),
+                new SymbolStatistics(LEFT_OTHER_COLUMN, UNKNOWN_STATS));
+        PlanNodeStatsEstimate rightStats = planNodeStats(
+                0,
```

Comment:
is row count 0 important? would 42 work?

Reply:
It can be 42. But than you need to set NDV to something, so the `calculateJoinComplementStats` doesn't fall into `return UNKNOWN_STATS` case. And then i would need to check the statistics, if they are correct. I decided to keep this test simple.